src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -94,6 +95,14 @@ public TimeSpan UserProcessorTime
             }
         }
 
+        public int HandleCount
+        {
+            get
+            {
+                return Directory.GetDirectories("/dev/fd", "*", SearchOption.TopDirectoryOnly).Length;
```

Comment:
I think we should avoid returning the number for the wrong process.  Either we should return something based on the target process, even if it's an approximation, or return a sentinel indicating we couldn't get a valid number (i.e. 0).  Maybe start with the latter and see if there's any reasonable way to do the former that's not too crazy.